cmake/kokkos_tribits.cmake

```diff
@@ -136,13 +136,22 @@ ENDFUNCTION()
 
 FUNCTION(KOKKOS_ADD_EXECUTABLE_AND_TEST ROOT_NAME)
 IF (KOKKOS_HAS_TRILINOS)
+  CMAKE_PARSE_ARGUMENTS(PARSE
+    ""
+    ""
+    "SOURCES;CATEGORIES;ARGS"
+    ${ARGN})
```

Comment:
We are now parsing the arguments in both branches. Let's move it before the if-switch (and also add `VERIFY_EMPTY(KOKKOS_ADD_EXECUTABLE_AND_TEST ${PARSE_UNPARSED_ARGUMENTS})` check).

Reply:
Yeah, good call.

Reply:
I moved the parsing outside of the if/else and remove the duplication, thanks.
The `VERIFY_EMPTY` check was in `fake_tribits` and was not found when I tried to call outside of the `ELSE` branch (that is within the `NOT KOKKOS_HAS_TRILINOS` branch), so I moved the `VERIFY_EMPTY` function to `kokkos_tribits`